packages/core/data/observable-array/index.ts

```diff
@@ -113,7 +114,11 @@ export class ObservableArray<T> extends Observable {
 
 	set length(value: number) {
 		if (types.isNumber(value) && this._array && this._array.length !== value) {
-			this.splice(value, this._array.length - value);
+			let added=[];
+			for (let i=this._array.length;i < value;++i) {
+				added.push(undefined);
+			}
```

Comment:
Would this be cleaner/faster? (According to a quick benchmark, it is 50% faster, but I don't trust benchmarks that much)

```suggestion
			let added = Array(Math.max(value - this._array.length, 0)).fill(undefined);
```

Reply:
Not sure on that one.   Interesting change...  I'm not opposed to it if you want to do another PR to change this...

The only thing is that we would have to probably put a comment on it; each time I look at it -- I have to read it several time to understand what it is doing.  :grinning:

Reply:
kind of agree with @NathanaelA. Also i am not even sure `Array.fill` is faster than `for`